src/main/java/at/hannibal2/skyhanni/features/inventory/bazaar/CraftMaterialsFromBazaar.kt

```diff
@@ -0,0 +1,166 @@
+package at.hannibal2.skyhanni.features.inventory.bazaar
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.features.inventory.bazaar.BazaarApi.Companion.isBazaarItem
+import at.hannibal2.skyhanni.utils.CollectionUtils.addOrPut
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalNameOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.itemName
+import at.hannibal2.skyhanni.utils.ItemUtils.itemNameWithoutColor
+import at.hannibal2.skyhanni.utils.ItemUtils.name
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUItems.getPrice
+import at.hannibal2.skyhanni.utils.NumberUtil
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class CraftMaterialsFromBazaar {
+
+    private val config get() = SkyHanniMod.feature.inventory.bazaar
+
+    private val matieralSlots = listOf(10, 11, 12, 19, 20, 21, 28, 29, 30)
+    private val pattern by RepoPattern.pattern(
+        "inventory.recipe.title",
+        ".* Recipe"
+    )
+
+    private var inRecipeInventory = false
+    private var purchasing = false
+    private var display = listOf<Renderable>()
+    private var neededMaterials = mapOf<NEUInternalName, Int>()
+    private var multiplier = 1
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        val correctInventoryName = pattern.matches(event.inventoryName)
+        val items = event.inventoryItems
+        val correctItem = items[23]?.name == "§aCrafting Table"
+
+        inRecipeInventory = correctInventoryName && correctItem && !purchasing
+        if (!inRecipeInventory) return
+
+        val recipeName = items[25]?.itemName ?: return
```

Comment:
why itemName instead of name directly?

Reply:
Name gives us the display name stored in the item stack itself. ItemName requests the name via neu repo and adds support for enchanted book and pet names.

Reply:
In general i think we should always use itemName when rendering, and only use name for comparing